### PR TITLE
Infrastructure to build Sphinx API documentation: Reloaded

### PR DESCRIPTION
**Pull request by valschneider**
Stealing @bjackman's https://github.com/ARM-software/lisa/pull/308

### CODE REVIEW

**Comment by valschneider** [comment]
@derkling @mdigiorgio @qperret Could you guys give it a try and see if the build goes well on your end ? I'd like this to get merged, it's been gathering dust for too long.

File: doc/requirements.txt [hunk 1]
```diff
@@ -4,3 +4,4 @@ devlib>=0.0.3
 psutil>=4.4.2
 wrapt>=1.10.8
 ipython>=4.2.0
+matplotlib>=1.4.2
```
Comment (mdigiorgio) [inline doc/requirements.txt]: Why isn't this in the previous commit?

File: .travis.yml [hunk 2]
```diff
@@ -19,8 +19,12 @@ sudo: required
 language: python
 install:
   - pip install --upgrade trappy bart-py devlib psutil wrapt matplotlib
+    wrapt jupyter matplotlib sphinx
```
Comment (mdigiorgio) [inline .travis.yml]: `wrapt` and `matplotlib` are added twice

File: doc/requirements.txt (hunk 1 above)
Comment (valschneider) [reply doc/requirements.txt]: Will fixup that bit, thanks for pointing it out.

**Comment by valschneider** [comment]
Took care of @mdigiorgio's changes.

**Pull request merged**